optuna/samplers/_nsga2/crossover.py

```diff
@@ -0,0 +1,397 @@
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import List
+from typing import Sequence
+
+import numpy as np
+
+from optuna._transform import _SearchSpaceTransform
+from optuna.distributions import BaseDistribution
+from optuna.distributions import CategoricalDistribution
+from optuna.distributions import DiscreteUniformDistribution
+from optuna.distributions import IntLogUniformDistribution
+from optuna.distributions import IntUniformDistribution
+from optuna.distributions import LogUniformDistribution
+from optuna.distributions import UniformDistribution
+from optuna.study import Study
+from optuna.study import StudyDirection
+from optuna.trial import FrozenTrial
+
+
+_NUMERICAL_DISTRIBUTIONS = (
+    UniformDistribution,
+    LogUniformDistribution,
+    DiscreteUniformDistribution,
+    IntUniformDistribution,
+    IntLogUniformDistribution,
+)
+
+
+def crossover(
```

Comment:
I'm a bit worried about the maintainability of this function and maybe more so the concrete crossover functions called from here. If you have time, it'd be great if you could add some explanatory comments to the algorithm operations, make variable names more readable, or elaborate/add on the references (I noticed some crossover algorithms were missing reference URLs.